worker/backend/container.go

```diff
@@ -1,146 +1,331 @@
+		return nil, fmt.Errorf("task: %w", err)
+	}
+
+	cioOpts := []cio.Opt{
+		cio.WithStreams(
+			processIO.Stdin,
+			processIO.Stdout,
+			processIO.Stderr,
+		),
+	}
+
+	proc, err := task.LoadProcess(ctx, pid, cio.NewAttach(cioOpts...))
+	if err != nil {
+		return nil, fmt.Errorf("load proc: %w", err)
+	}
+
+	status, err := proc.Status(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("proc status: %w", err)
+	}
+
+	if status.Status != containerd.Running {
+		return nil, fmt.Errorf("proc not running: status = %s", status.Status)
+	}
+
+	exitStatusC, err := proc.Wait(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("proc wait: %w", err)
+	}
+
+	return NewProcess(proc, exitStatusC), nil
+}
 
 // Properties returns the current set of properties
-func (c *Container) Properties() (properties garden.Properties, err error) { return }
+//
+func (c *Container) Properties() (garden.Properties, error) {
+	ctx := context.Background()
+
+	labels, err := c.container.Labels(ctx)
+	if err != nil {
+		return garden.Properties{}, fmt.Errorf("")
+	}
+
+	return labels, nil
+}
 
 // Property returns the value of the property with the specified name.
 //
-// Errors:
-// * When the property does not exist on the container.
-func (c *Container) Property(name string) (value string, err error) { return }
+func (c *Container) Property(name string) (string, error) {
+	properties, err := c.Properties()
+	if err != nil {
+		return "", err
+	}
+
+	v, found := properties[name]
+	if !found {
+		return "", ErrNotFound(name)
+	}
+
+	return v, nil
+}
 
 // Set a named property on a container to a specified value.
 //
-// Errors:
-// * None.
-func (c *Container) SetProperty(name string, value string) (err error) { return }
+func (c *Container) SetProperty(name string, value string) error {
+	labelSet := map[string]string{
+		name: value,
+	}
 
-// Remove a property with the specified name from a container.
-//
-// Errors:
-// * None.
-func (c *Container) RemoveProperty(name string) (err error) { return }
+	_, err := c.container.SetLabels(context.Background(), labelSet)
+	if err != nil {
+		return fmt.Errorf("set label: %w", err)
+	}
+
+	return nil
+}
+
+// RemoveProperty - Not Implemented
+func (c *Container) RemoveProperty(name string) (err error) {
+	err = ErrNotImplemented
+	return
+}
+
+// Info - Not Implemented
+func (c *Container) Info() (info garden.ContainerInfo, err error) {
+	err = ErrNotImplemented
+	return
+}
+
+// Metrics - Not Implemented
+func (c *Container) Metrics() (metrics garden.Metrics, err error) {
+	err = ErrNotImplemented
+	return
+}
+
+// StreamIn - Not Implemented
+func (c *Container) StreamIn(spec garden.StreamInSpec) (err error) {
+	err = ErrNotImplemented
+	return
+}
+
+// StreamOut - Not Implemented
+func (c *Container) StreamOut(spec garden.StreamOutSpec) (readCloser io.ReadCloser, err error) {
+	err = ErrNotImplemented
+	return
+}
+
+// SetGraceTime - Not Implemented
+func (c *Container) SetGraceTime(graceTime time.Duration) (err error) {
```

Comment:
We actually use this during gc for hijacked containers

Reply:
https://github.com/concourse/concourse/issues/5134